website/content/projects/2019/spec.md

```diff
@@ -118,11 +118,62 @@ The project will span seven weeks (Weeks 4 - 10), with a final review and team p
 *   Weeks 9-10: Sprint 4
 
 
-<!--
+
 ## Suggested initial goals
 
-TODO
--->
+To get you started, we suggest each team focus on the following set of initial goals, most of which won't require much communication with other teams at first (allowing you to focus on the requirements of your component before you have to start collaborating with other teams)
+
+### WDL
+
+We don't expect you to define a language completely from scratch, and suggest that you use [YAML](https://en.wikipedia.org/wiki/YAML) as the base format for the WDL. We have even provided some sample WDL files to get you started. However, you shouldn't regard these files as normative: they are a suggestion on what a WDL file would look like, but it is up to you to decide the exact format of the files. Furthermore, while YAML files can be parsed in C using the [libyaml](https://github.com/yaml/libyaml) library, we have provided a library called libobj that provides an abstraction over the low-level operations provided by libyaml.
+
+So, the WDL team can initially focus on this:
+
+- Writing a draft specification of the WDL language
+- Writing a module for loading WDL files. This module must do some basic validation of the file (ensuring it meets the specification of the WDL language)
+
+After this, you will find you will need to work more closely with the Game State team to figure out how to load the contents of the WDL file into the data structures provided by Game State.
+
+
+### Game State
+
+This team will be responsible for designing the data structures that will store the state of the game in memory, as well as a module to manipulate those data structures. A good initial goal is to focus on *designing* these modules by writing the modules' header files without yet writing the implementation of the functions promised by the module's interface. This will get you thinking about what data structures to use, how they will relate to other data structures, what operations you want to allow on each data structure.
```

Comment:
We should tell them to do an additional phase of design before writing headers - perhaps UML, perhaps just unstructured whiteboarding.

Reply:
We should also tell them that they need to provide a nicely readable format of their design - whether that's UML, plain text, or something else is up to them, but I think they should provide a visually easy reference for the other teams.

Reply:
Unfortunately, we haven't covered any UML this year. Suggesting that they do whiteboarding sounds good to me; any suggestions on how to get them to produce a "visually easy reference" (maybe just a wiki page in the repository?)

Reply:
Yeah I think that would work

Reply:
I agree that a wiki page would be good, or alternatively, a README in the game state directory. We should just keep it consistent for all the teams so everyone knows where to look, so for example, action management should maybe put their specification in a wiki too

Reply:
I'd lean towards READMEs because wikis don't undergo code review

Reply:
How about having a top-level `docs/` directory that contains both user-facing documentation, and developer-facing documentation? (this could eventually be incorporated into a chiventure website)

Reply:
I like that, yeah

Reply:
I've created an issue in the chiventure repository to discuss this further: https://github.com/uchicago-cs/chiventure/issues/22